File: Assets/Scripts/OSC/OSCPeakRotateMulti.js
var oscPeakObjects : OSCPeakRotateObject[];


class OSCPeakRotateObject {
	public var peakObject : GameObject;
	public var channel : int = 1;
	public var minVector : Vector3;
	public var maxVector : Vector3;
}


function OSCAudioEnvelope (input) {
	for(oscPeakObject in oscPeakObjects) {
		if(input.channel == oscPeakObject.channel) {
			oscPeakObject.peakObject.transform.localEulerAngles =  Vector3.Lerp(oscPeakObject.minVector, oscPeakObject.maxVector, input.envelope);
		}
	}
}

File: Assets/Scripts/OSC/OSCPeakRotateMulti.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

public class OSCPeakRotateMulti : MonoBehaviour 
{
	public List<OSCPeakRotateObject> oscPeakObjects;
	
	[System.Serializable]
	public class OSCPeakRotateObject {
		public GameObject peakObject;
		public int channel = 1;
		public Vector3 minVector;
		public Vector3 maxVector;
	} 
	
	
	void OSCAudioEnvelope (OSCHandler.AudioEnvelopeData input) {
		foreach(OSCPeakRotateObject oscPeakObject in oscPeakObjects) {
			if(input.channel == oscPeakObject.channel) {
				oscPeakObject.peakObject.transform.localEulerAngles =  Vector3.Lerp(oscPeakObject.minVector, oscPeakObject.maxVector, input.envelope);
			}
		}
	}
}
